src/app/services/messages.service.ts

```diff
@@ -23,4 +23,8 @@ export class MessagesService {
       this.toastrService.error('Something went wrong! Please try again later');
     }
   }
+
+  public handlerWarning() {
```

Comment:
You use this warning only for gender validation, please rename to a more specific method, or (better) make this method reusable for other warnings in future.